back/prisma/migrations/64_appendix2_fraction.sql

```diff
@@ -0,0 +1,39 @@
+-- CreateTable
+CREATE TABLE "FormGroupement" (
+    "id" TEXT NOT NULL,
+    "nextFormId" TEXT NOT NULL,
+    "initialFormId" TEXT NOT NULL,
+    "quantity" DOUBLE PRECISION NOT NULL,
+
+    CONSTRAINT "FormGroupement_pkey" PRIMARY KEY ("id")
+    CONSTRAINT "FormGroupement_pkey_unique_together" UNIQUE ("nextFormId", "initialFormId")
+);
+
+-- CreateIndex
+CREATE INDEX "_FormGroupementNextFormId" ON "FormGroupement"("nextFormId");
+
+-- CreateIndex
+CREATE INDEX "_FormGroupementInitialFormId" ON "FormGroupement"("initialFormId");
+
+-- AddForeignKey
+ALTER TABLE "FormGroupement" ADD CONSTRAINT "FormGroupement_nextFormId_fkey" FOREIGN KEY ("nextFormId") REFERENCES "Form"("id") ON DELETE RESTRICT ON UPDATE CASCADE;
+
+-- AddForeignKey
+ALTER TABLE "FormGroupement" ADD CONSTRAINT "FormGroupement_initialFormId_fkey" FOREIGN KEY ("initialFormId") REFERENCES "Form"("id") ON DELETE RESTRICT ON UPDATE CASCADE;
+
+-- Migrate data
+INSERT INTO "default$default"."FormGroupement" ("initialFormId", "nextFormId", "quantity")
```

Comment:
l 'ID n'étant pas auto-généré ça devrait planter

Reply:
Je ne sais pas trop quelle stratégie adopter ici du coup. Si je veux des identifiants prisma type `cuid` je dois gérer ça avec un script mais je ne peux pas gérer la suppression de `appendix2RootFormId` dans la même PR. Il faudra penser à les supprimer ultérieurement.

Reply:
je pense qu'on peut insérer des UUID  postgre ? quitte à scripter un UUID -> cuid après

Reply:
du style (avec la gestion de quantityReceived) : `INSERT INTO "default$default"."FormGroupement" ("id", "initialFormId", "nextFormId", "quantity")
SELECT gen_random_uuid(), "id", "appendix2RootFormId", COALESCE("quantityReceived",0) FROM "default$default"."Form" WHERE "appendix2RootFormId" IS NOT NULL;`

Reply:
Ah ouais c'est une bonne idée. J'ai pris le parti pour l'instant d'écrire un script et de supprimer `Form.appendix2FormId` dans une migration ultérieure. Ça rend la migration un peu plus testable.